openapi.go

```diff
@@ -47,35 +58,79 @@ func (s *Server) GenerateOpenAPI() {
 	}
 
 	// Write spec to docs/openapi.json
-	err = os.MkdirAll("docs", 0o750)
-	if err != nil {
-		slog.Error("Error creating docs directory", "error", err)
-	}
-	f, err := os.Create("docs/openapi.json")
-	if err != nil {
-		slog.Error("Error creating docs/openapi.json", "error", err)
+	if options.disableSaveOnDisk == false {
```

Comment:
The fact of having 2 large, hyper-long `if` is a bit of an indicator of poor code structure. Personally, I'd refactor it into 2 new functions that are called or not depending on whether the option is called or not. What do you think?